src/core/channels/channel.ts

```diff
@@ -96,7 +96,8 @@ export default class Channel extends EventsDispatcher {
     this.subscriptionCancelled = false;
     this.authorize(this.pusher.connection.socket_id, (error, data)=> {
       if (error) {
-        this.emit('pusher:subscription_error', data)
+        Logger.error(data);
```

Comment:
I think it would be nicer to bind to this event from a level above, and log. The reason I didn't was because I didn't want to change the behaviour so that `No callbacks on <channel> for pusher:subscription_error` would no longer be logged. See:

https://github.com/pusher/pusher-js/blob/cac7255980a74977bbe40365fc161579deeb945a/src/core/channels/channel.ts#L27

I guess I could duplicate this log line in the handler for `pusher:subscription_error`, but I'm not sure that's much better.

Reply:
I think that was my thinking. It is a good point that this might make it *less* obvious that you can `bind` to error events.

Reply:
I don't feel too strongly about it. I left it as it is. I added a comment to explain why it is logged here.